shift_detector/precalculations/TextMetadata.py

```diff
@@ -54,6 +52,97 @@ def process(self, store):
             metadata2[column] = [self.metadata_function(text) for text in clean2]
         return metadata1, metadata2
 
+class GenericTextMetadataWithTokenizing(GenericTextMetadata):
+
+    @staticmethod
+    @abstractmethod
+    def metadata_name() -> str:
+        pass
+
+    @abstractmethod
+    def metadata_column_type(self) -> ColumnType:
+        pass
+
+    @abstractmethod
+    def metadata_function(self, words):
+        pass
+
+    def process(self, store):
+        metadata1 = pd.DataFrame()
+        metadata2 = pd.DataFrame()
+        df1, df2 = store[TokenizeIntoWords()]
+        for column in df1.columns:
+            logger.info(self.metadata_name() + ' analysis for ' + column)
+            metadata1[column] = [self.metadata_function(words) for words in df1[column]]
+            metadata2[column] = [self.metadata_function(words) for words in df2[column]]
+        return metadata1, metadata2
+
+class GenericTextMetadataWithTokenizingAndLanguage(GenericTextMetadata):
+
+    def __init__(self, language='en', infer_language=False):
+        self.language = language
+        self.infer_language = infer_language
+
+    @staticmethod
+    @abstractmethod
+    def metadata_name() -> str:
+        pass
+
+    @abstractmethod
+    def metadata_column_type(self) -> ColumnType:
+        pass
+
+    @abstractmethod
+    def metadata_function(self, words):
+        pass
+
+    def process(self, store):
+        metadata1 = pd.DataFrame()
+        metadata2 = pd.DataFrame()
+        df1, df2 = store[TokenizeIntoWords()]
+        if self.infer_language:
+            lang1, lang2 = store[LanguageMetadata()]
+        for column in df1.columns:
+            logger.info(self.metadata_name() + ' analysis for ' + column)
+            temp_column1 = []
+            temp_column2 = []
+            for i in range(len(df1)):
+                if self.infer_language:
```

Comment:
put line lang1, lang2 = store[LanguageMetadata()] in here?

Reply:
would not make sense to do that in every iteration of the for loop